### PR TITLE
Explain how to install and configure curator

### LINKED ISSUE #934: [ELK] It should be explained how to use and install curator
https://app.intercom.com/a/apps/w4oogu7s/inbox/inbox/all/conversations/26758479101

### PR DESCRIPTION
**Pull request by johnsudaar**
Fix #934

### CODE REVIEW

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 1]
```diff
@@ -234,5 +234,117 @@ output {
...
+```
+ELASTICSEARCH_HOST=host:port
+ELASTICSEARCH_AUTH=user:password
+```
+
+The last step is to configure your indices life cycle. This is based on your index names.
+When you configured logstash, we asked you to configure your index with the following name `sc-apps-%{+YYYY.MM.dd}`.
+We will configure Curator to parse the date contained on those names and delete the ones that are too old.
+
+```
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:315]:
```suggestion
```yaml
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 2]
```diff
@@ -234,5 +234,117 @@ output {
...
+      value: ${LOGS_INDICES_PREFIX}
+    - filtertype: age
+      source: name
+      direction: older
+      timestring: '%Y.%m.%d'
+      unit: days
+      unit_count: ${LOGS_RETENTION_DAYS}
+```
+
+You will now need to add two environnement variables:
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:335]:
```suggestion
You now need to add two environment variables:
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 3]
```diff
@@ -234,5 +234,117 @@ output {
...
+```
+LOGS_INDICES_PREFIX=sc-apps
+```
+
+This variable let you filter index that should be affected by this policy (so you wont delete other index that are stored in the same database);
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:340]:
```suggestion
This variable lets you filter index that should be affected by this policy (so you wont delete other indices that are stored in the same database):
```

The parenthesis does not look relevant. Isn't it just a sentence?

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 4]
```diff
@@ -234,5 +234,117 @@ output {
...
+
+```
+LOGS_RETENTION_DAYS=10
+```
+
+This variable let you configure the retention time of your logs. Here it will delete an index if it is 10+ days old.
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:346]:
```suggestion
This variable lets you configure the retention time of your logs. With this configuration, Curator will delete an index if it is 10+ days old.
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 5]
```diff
@@ -234,5 +234,117 @@ output {
...
+  logfile:
+  logformat: default
+```
+
+Sadly curator cannot use the `ELASTICSEARCH_URL` you will need to define 2 other environnement variable on your app:
+
+So if your `ELASTICSEARCH_URL` is set to `http://user:password@host:port`, you'll need to define 2 environnement variables:
+
+```
+ELASTICSEARCH_HOST=host:port
```
[the remark this answers is not in the document]
Comment (johnsudaar) [reply _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:304]: They need to add use_ssl and other cert stuff. Not sure we want to add this to this page. It's already pretty long...

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 6]
```diff
@@ -234,5 +234,114 @@ output {
 }
 ```
 
-This tutorial is based on an article published on [our
-blog](https://scalingo.com/articles/2018/02/23/running-the-elk-stack-on-scalingo.html).
+## Curator
+
+### Installing Curator
+
+Keeping all your application logs is not always feasible due to the huge amount of storage that would be needed. However logs are usually only relevant for a short period of time. That's why it's current practice to destroy documents if they get too old.
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:241]:
```suggestion
Keeping all your application logs is not always feasible due to the huge amount of storage that would be needed. However logs are usually only relevant for a short period of time. That's why the current best practice is to destroy logs after a certain amount of time.
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 7]
```diff
@@ -234,5 +234,114 @@ output {
...
+
+This is where [Curator](https://www.elastic.co/guide/en/elasticsearch/client/curator/5.8/index.html) is needed. This project is designed to let you managed your indices life cycle.
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:243]:
```suggestion
This is where [Curator](https://www.elastic.co/guide/en/elasticsearch/client/curator/5.8/index.html) is needed. This project is designed to let you manage your indices life cycle.
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 8]
```diff
@@ -234,5 +234,114 @@ output {
...
+```
+
+In order to instruct the Python buildpack to install Curator and its dependencies, create a file named `requirements.txt` at the root of your project:
+
+```
+PyYAML==5.3.1
+elasticsearch-curator==5.8.0
+```
+
+Finally you'll need a file to write a script that will run curator every 12h. This can be achieved via a simple bash script.
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:261]:
```suggestion
Finally you need to run Curator every 12h. This can be achieved via a shell script:
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 9]
```diff
@@ -234,5 +234,114 @@ output {
...
+
+while true; do
+  echo "Running Curator"
+  curator --config curator.yml log-clean.yml
+  # Run Curator every 12h
+  sleep 43200
+done
+```
+
+Finally you need to tell Scalingo to start Curator.
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:276]:
```suggestion
Finally you need to instruct Scalingo to start Curator.
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 10]
```diff
@@ -234,5 +234,114 @@ output {
...
+  hosts:
+    - ${ELASTICSEARCH_HOST}
+  http_auth: ${ELASTICSEARCH_AUTH}
+logging:
+  loglevel: INFO
+  logfile:
+  logformat: default
+```
+
+Curator cannot use the `ELASTICSEARCH_URL`. You need to define 2 other environment variables on your app duplicating `ELASTICSEARCH_URL` content.
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:300]:
```suggestion
Curator cannot use the `ELASTICSEARCH_URL` environment variable. You need to define two other environment variables on your app, duplicating `ELASTICSEARCH_URL` content.
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 11]
```diff
@@ -234,5 +234,114 @@ output {
...
+      value: ${LOGS_INDICES_PREFIX}
+    - filtertype: age
+      source: name
+      direction: older
+      timestring: '%Y.%m.%d'
+      unit: days
+      unit_count: ${LOGS_RETENTION_DAYS}
+```
+
+You now need to add two environment variables:
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:332]: It's kinda odd to read "two environment variables" but actually only seeing one. I would rephrase to prevent this odd thing. Moreover you don't explain what this environment variable should contain (even though the variable name is almost self-explanatory).

Suggestion:
```suggestion
You now need to add two environment variables. The first environment variable is `LOGS_INDICES_PREFIX`. It configures the index pattern that should be affected by this policy. Setting this variable to `sc-apps` prevent Curator from deleting the other indices that are stored in the same database.

The second environment variable is `LOGS_RETENTION_DAYS`. It configures the retention time of your logs (in days). Setting this variable to `10`, Curator will delete an index if it is 10+ days old.

```
LOGS_INDICES_PREFIX=sc-apps 
LOGS_RETENTION_DAYS=10 
```
```

If this suggestion suits you, you can remove the text below.

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 12]
```diff
@@ -234,5 +234,114 @@ output {
...
+    - ${ELASTICSEARCH_HOST}
+  http_auth: ${ELASTICSEARCH_AUTH}
+logging:
+  loglevel: INFO
+  logfile:
+  logformat: default
+```
+
+Curator cannot use the `ELASTICSEARCH_URL`. You need to define 2 other environment variables on your app duplicating `ELASTICSEARCH_URL` content.
+Hence if your `ELASTICSEARCH_URL` variable is set to `http://user:password@host:port`, you need to define 2 environment variables:
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:301]:
```suggestion
Hence if your `ELASTICSEARCH_URL` variable is set to `http://user:password@host:port`, you need to define two environment variables:
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 13]
```diff
@@ -234,5 +234,136 @@ output {
...
+project:
+
+```
+PyYAML==5.3.1
+elasticsearch-curator==5.8.0
+```
+
+
+Curator is not a deamon, it is designed as a one-off process. To be able
+to run it on Scalingo you'll need to write a simple bash script that launch
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:270]: Nothing is "simple" :smiling_imp: 

```suggestion
to run it on Scalingo you need to write a Bash script that executes
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 14]
```diff
@@ -234,5 +234,136 @@ output {
...
+Curator regularly. This can be achieved via a simple shell script.
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:271]:
```suggestion
Curator regularly.
```

You already wrote that

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 15]
```diff
@@ -234,5 +234,136 @@ output {
...
+
+```
+ELASTICSEARCH_HOST=host:port
+ELASTICSEARCH_AUTH=user:password
+```
+
+The last step is to configure your indices life cycle. This is based on your
+indices names.
+When you configured Logstash, we asked you to configure your index with the
+following name `sc-apps-%{+YYYY.MM.dd}`.
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:326]: I think this sentence is not correctly placed. We don't really understand why you remind us about it here. It is actually related to the value of `LOGS_INDICES_PREFIX` you explain a few lines later. Maybe move this sentence later?

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 16]
```diff
@@ -234,5 +234,136 @@ output {
...
+Create a file named: `log-clean.yml`. This configuration parse the indices
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:327]:
```suggestion
Create a file named `log-clean.yml`. This configuration parses the indices
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md [hunk 17]
```diff
@@ -234,5 +234,136 @@ output {
...
+names stored in elasticsearch and remove the one that too old.
```
Comment (EtienneM) [inline _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:328]:
```suggestion
names stored in Elasticsearch and removes the ones that are too old.
```

File: _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md (hunk 15 above)
Comment (EtienneM) [reply _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:326]: You didn't addressed my question here I think

Comment (johnsudaar) [reply _posts/platform/getting-started/2000-01-01-getting-started-with-elk.md:326]: Or just remove it.
